src/librustc/mir/visit.rs

```diff
@@ -675,20 +675,12 @@ macro_rules! make_mir_visitor {
                             place: & $($mutability)? Place<'tcx>,
                             context: PlaceContext,
                             location: Location) {
-                match place {
-                    Place::Base(place_base) => {
-                        self.visit_place_base(place_base, context, location);
-                    }
-                    Place::Projection(proj) => {
-                        let context = if context.is_mutating_use() {
-                            PlaceContext::MutatingUse(MutatingUseContext::Projection)
-                        } else {
-                            PlaceContext::NonMutatingUse(NonMutatingUseContext::Projection)
-                        };
+                let Place { base, projection } = place;
 
-                        self.visit_place(& $($mutability)? proj.base, context, location);
```

Comment:
just sanity checking: have you gone through all `visit_place` implementations in rustc and checked whether they should manually iterate?

Reply:
Yes, have done that.